integration/volume/volume_test.go

```diff
@@ -234,3 +236,46 @@ func TestVolumePruneAnonymous(t *testing.T) {
 	assert.Check(t, cmp.Contains(pruneReport.VolumesDeleted, v.Name))
 	assert.Check(t, cmp.Contains(pruneReport.VolumesDeleted, vNamed.Name))
 }
+
+func TestVolumePruneAnonFromImage(t *testing.T) {
+	defer setupTest(t)()
+	client := testEnv.APIClient()
+
+	dockerfile := `FROM busybox
+VOLUME /foo
+`
+
+	buildContext := fakecontext.New(t, "", fakecontext.WithDockerfile(dockerfile))
+	img := "busybox:withvolume"
+
+	ctx := context.Background()
+	resp, err := client.ImageBuild(ctx, buildContext.AsTarReader(t), types.ImageBuildOptions{
+		Tags: []string{img},
+	})
+	assert.NilError(t, err)
+
+	_, err = io.Copy(io.Discard, resp.Body)
+	resp.Body.Close()
+	assert.NilError(t, err)
```

Comment:
Does this need the ugly hack we use elsewhere to check if the image was actually built? (I have a feeling there's still other tests that don't check for this, because of the kinda horrible streaming response); https://github.com/moby/moby/blob/2fa66cfce2523f6e38bcccc1161055b46c3c6a90/integration/build/build_test.go#L521-L524